src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -149,7 +169,24 @@ private string ResolveAttributeLeasesConnectionString(CosmosDBTriggerAttribute a
 
         private ChangeFeedHostOptions ResolveLeaseOptions(CosmosDBTriggerAttribute attribute)
         {
-            return attribute.LeaseOptions ?? _leasesOptions;
+            ChangeFeedHostOptions triggerChangeFeedHostOptions = new ChangeFeedHostOptions();
+            triggerChangeFeedHostOptions.LeasePrefix = attribute.LeaseCollectionPrefix ?? _leasesOptions.LeasePrefix;
+            triggerChangeFeedHostOptions.FeedPollDelay = ResolveTimeSpanFromMilliseconds(_leasesOptions.FeedPollDelay, attribute.FeedPollDelay);
+            triggerChangeFeedHostOptions.LeaseAcquireInterval = ResolveTimeSpanFromMilliseconds(_leasesOptions.LeaseAcquireInterval, attribute.LeaseAcquireInterval);
+            triggerChangeFeedHostOptions.LeaseExpirationInterval = ResolveTimeSpanFromMilliseconds(_leasesOptions.LeaseExpirationInterval, attribute.LeaseExpirationInterval);
+            triggerChangeFeedHostOptions.LeaseRenewInterval = ResolveTimeSpanFromMilliseconds(_leasesOptions.LeaseRenewInterval, attribute.LeaseRenewInterval);
+            triggerChangeFeedHostOptions.CheckpointFrequency = _leasesOptions.CheckpointFrequency ?? new CheckpointFrequency();
+            if (attribute.CheckpointFrequency > 0)
```

Comment:
would it ever make sense to set either of these checkpoint properties to 0?

Reply:
No, any 0 interval would probably send the renewal threads running all the time instead of their usual backoff.